Android/Shared/src/main/java/com/bentley/sample/shared/ActivityTimer.kt

```diff
@@ -0,0 +1,96 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+package com.bentley.sample.shared
+
+import com.github.itwin.mobilesdk.ITMLogger
+import java.lang.Integer.max
+import java.text.SimpleDateFormat
+import java.util.*
+
+/**
+ * Utility to record the timestamp of checkpoints and then log how long all the checkpoints took.
+ */
+class ActivityTimer(private val nameTitle: String = "ACTIVITY") {
+    var enabled = true
+    private val startTime = Date()
+    private var checkpoints: Array<Pair<String, Date>> = emptyArray()
+    private val maxLengths: Array<Int> = Array(4) { 0 }
+
+    /**
+     * Add a checkpoint
+     * @param name: The name of the checkpoint
+     */
+    fun addCheckpoint(name: String) {
+        if (!enabled) return
+        checkpoints += Pair(name, Date())
+    }
+
+    /**
+     * Create a string representing the elapsed time (in seconds and milliseconds) between two [Date] values.
+     * @param start: The start date for the elapsed time
+     * @param end: The end date for the elapsed time
+     * @return A string representing the elapsed time between the two dates
+     */
+    private fun timeDelta(start: Date, end: Date): String {
+        val delta = end.time - start.time
+        return "${delta / 1000}.${(delta % 1000).toString().padStart(3, '0')}"
```

Comment:
I think something like this should work: `String.format("%.3f", delta / 1000)`

Reply:
Fixed.